internal/a3m/client.go

```diff
@@ -15,11 +13,10 @@ type Client struct {
 	TransferClient transferservicev1beta1grpc.TransferServiceClient
 }
 
-func NewClient(ctx context.Context, tp trace.TracerProvider, addr string) (*Client, error) {
+func NewClient(tp trace.TracerProvider, addr string) (*Client, error) {
 	c := &Client{}
 
-	conn, err := grpc.DialContext(
-		ctx,
+	conn, err := grpc.NewClient(
```

Comment:
Would this be enough? Should we check the connection? Reading the links in the previous update it seems unnecessary. And, should we close the connection after all?

Reply:
@jraddaoui it looks like `defer conn.Close()` is expected:
https://github.com/grpc/grpc-go/blob/master/examples/helloworld/greeter_client/main.go#L49

But other than that I think your change above is sufficient.